ping/rust/src/lib.rs

```diff
@@ -31,30 +31,37 @@ pub async fn run_ping<S>(mut swarm: S) -> Result<()>
 where
     S: PingSwarm,
 {
-    info!("Running ping test: {}", swarm.local_peer_id());
-
     env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
 
-    let client = testground::client::Client::new_and_init().await.unwrap();
+    info!("Running ping test: {}", swarm.local_peer_id());
 
-    let local_addr = match if_addrs::get_if_addrs()
-        .unwrap()
+    let client = testground::client::Client::new_and_init()
+        .await
+        .map_err(|e| anyhow::anyhow!(e.to_string()))
+        .context("Unable to init testground cient.")?;
```

Comment:
```suggestion
        .context("Unable to init testground cient.")?;
```

Double mapping is a bit useless :)

Reply:
I'll look at this in the morning (diff TZ than you), but the suggestion caused compilation errors:
```
error[E0599]: the method `context` exists for enum `Result<Client, Box<dyn std::error::Error>>`, but its trait bounds were not satisfied
   --> src/lib.rs:41:10
    |
41  |           .context("Unable to init testground cient.")?;
    |            ^^^^^^^ method cannot be called on `Result<Client, Box<dyn std::error::Error>>` due to unsatisfied trait bounds
    |
   ::: .rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/result.rs:504:1
    |
504 |   pub enum Result<T, E> {
    |   --------------------- doesn't satisfy `_: anyhow::Context<Client, Box<dyn std::error::Error>>`
    |
   ::: .rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/boxed.rs:194:1
    |
194 | / pub struct Box<
195 | |     T: ?Sized,
196 | |     #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
197 | | >(Unique<T>, A);
    | | -
    | | |
    | | doesn't satisfy `Box<dyn std::error::Error>: Sync`
    | |_doesn't satisfy `Box<dyn std::error::Error>: std::marker::Send`
    |   doesn't satisfy `_: anyhow::context::ext::StdError`
    |
    = note: the following trait bounds were not satisfied:
            `Box<dyn std::error::Error>: anyhow::context::ext::StdError`
            which is required by `Result<Client, Box<dyn std::error::Error>>: anyhow::Context<Client, Box<dyn std::error::Error>>`
            `Box<dyn std::error::Error>: std::marker::Send`
            which is required by `Result<Client, Box<dyn std::error::Error>>: anyhow::Context<Client, Box<dyn std::error::Error>>`
            `Box<dyn std::error::Error>: Sync`
            which is required by `Result<Client, Box<dyn std::error::Error>>: anyhow::Context<Client, Box<dyn std::error::Error>>`
```

Reply:
Ah damn, that trait object is not `Send + 'static`! I think it would work otherwise. That is a shame. Let's leave it then!